### PR TITLE
♻️  Reworked spacer to match IOS behavior

### PR DESCRIPTION
**Pull request by andretortolano**
first time around I thought spacer would take width/height, but instead it seems like it uses "spacing" as a way for determining space it takes depending on which scope its in (row or column) this PR changes so it reflects IOS behavior

### CODE REVIEW

File: appcues/src/main/java/com/appcues/data/mapper/step/primitives/SpacerPrimitiveMapper.kt [hunk 1]
```diff
@@ -1,10 +1,9 @@
 package com.appcues.data.mapper.step.primitives
 
-import com.appcues.data.mapper.styling.mapComponentStyle
 import com.appcues.data.model.ExperiencePrimitive.SpacerPrimitive
 import com.appcues.data.remote.response.step.primitive.PrimitiveResponse.SpacerPrimitiveResponse
 
 internal fun SpacerPrimitiveResponse.mapSpacerPrimitive() = SpacerPrimitive(
     id = id,
-    style = style.mapComponentStyle(),
+    spacing = spacing,
```
Comment (iujames) [inline appcues/src/main/java/com/appcues/data/mapper/step/primitives/SpacerPrimitiveMapper.kt:8]: I see this matches iOS now https://github.com/appcues/appcues-ios-sdk/blob/main/Sources/AppcuesKit/Data/Models/Experience/ExperienceComponent.swift#L238

Looks like we should update the spec to include the `spacing` property? https://github.com/appcues/appcues-mobile-experience-spec/blob/main/experience-schema.json#L480-L490

File: appcues/src/main/java/com/appcues/ui/primitive/HorizontalStackPrimitive.kt [hunk 2]
```diff
@@ -60,9 +66,8 @@ private fun RowScope.ItemBox(
             .align(primitive.style.getVerticalAlignment(parentVerticalAlignment))
             .then(
                 if (distribution == ComponentDistribution.EQUAL ||
-                    // for when its a spacer with no width defined,
-                    // we will also fill available space
-                    (primitive is SpacerPrimitive && primitive.style.width == null)
+                    // for when its a spacer with no set spacing, it will fill available space
+                    (primitive is SpacerPrimitive && primitive.spacing.eq(0.0))
```
Comment (iujames) [inline appcues/src/main/java/com/appcues/ui/primitive/HorizontalStackPrimitive.kt:70]: just noting, can use `eq` as infix as well
```
(primitive is SpacerPrimitive && primitive.spacing eq 0.0)
```

**Review by iujames — APPROVED** [review]
The `StackScope` makes sense to me - applies a similar sort of behavior to what is built into SwiftUI for expanding across the axis of the containing stack layout 👍

File: appcues/src/main/java/com/appcues/data/mapper/step/primitives/SpacerPrimitiveMapper.kt (hunk 1 above)
Comment (andretortolano) [reply appcues/src/main/java/com/appcues/data/mapper/step/primitives/SpacerPrimitiveMapper.kt:8]: yes

**Pull request merged**